aten/src/ATen/cpu/vec256/vec256_bfloat16.h

```diff
@@ -203,7 +203,23 @@ template <> class Vec256<BFloat16> {
     return cvtfp32_bf16(o1, o2);
   }
   Vec256<BFloat16> angle() const {
-    return _mm256_set1_epi16(0);
+    __m256 lo, hi;
+    cvtbf16_fp32(values, lo, hi);
+    auto angle_lambda = [](__m256 values) {
+      const auto zero_vec = _mm256_set1_ps(0.f);
+      const auto nan_vec = _mm256_set1_ps(NAN);
+      const auto not_nan_mask = _mm256_cmp_ps(values, values, _CMP_EQ_OQ);
```

Comment:
sorry I am not very well versed with the AVX instructions ... what (nan related) information do we get by comparing values to itself?

Reply:
Since AVX itself does not have a special function for masking `nan`s.
We use the property `nan != nan` to find the `nan` values.